src/production/__tests__/CodeSplitting.test.ts

```diff
@@ -0,0 +1,347 @@
+      'bar.ts': ``,
+    });
+    const { modules, splitEntries: { entries } } = context;
+    expect(modules).toHaveLength(8);
+    expect(entries).toHaveLength(3);
+    expect(entries[0].entry.getShortPath()).toEqual('default/one.js');
+    expect(entries[0].modules).toHaveLength(2);
+    expect(entries[1].entry.getShortPath()).toEqual('default/two.js');
+    expect(entries[1].modules).toHaveLength(1);
+    expect(entries[2].entry.getShortPath()).toEqual('default/three.js');
+    expect(entries[2].modules).toHaveLength(2);
+    expect(modules[7].isCommonsEligible()).toBe(true);
+  });
+
+  it('should have flag commons correctly', async () => {
+    const context = await test({
+      'index.ts': `
+        import './foo';
+        async function load() {
+          await import('./bar');
+          await import('./one');
+        }
+      `,
+      'foo.ts': `
+        export default function() {}
+      `,
+      'bar.ts': `
+        import './utils';
+        import './foo';
+        export default function() {}
+      `,
+      'one.ts': `
+        import './utils';
+      `,
+      'utils.ts': ``,
+    });
+    const { modules, splitEntries: { entries } } = context;
+    expect(modules).toHaveLength(5);
+    expect(entries).toHaveLength(2);
+    // utils.ts
+    expect(modules[2].isCommonsEligible()).toBe(true);
+    // foo.ts
+    expect(modules[4].isCommonsEligible()).toBe(false);
+  });
+
+  it('should detect circular dependency', async () => {
+    const context = await test({
+      'index.ts': `
+        async function load() {
+          const dynamicFunc = await import('./a');
+        }
+      `,
+      'a.ts': `
+        import './b';
+        export default function() {};
+      `,
+      'b.ts': `
+        import './c';
+      `,
+      'c.ts': `
+        import './b';
+      `,
+    });
+    expect(context.modules).toHaveLength(4);
+    // we only have 1 split
+    expect(context.splitEntries.entries).toHaveLength(1);
+    // the split we have has 3 modules: a, b, c
+    expect(context.splitEntries.entries[0].modules).toHaveLength(3);
+  });
+
+  it('should detect deep circular dependency', async () => {
+    const context = await test({
+      'index.ts': `
+        async function load() {
+          const dynamicFunc = await import('./a');
+        }
+      `,
+      'a.ts': `
+        import './b';
+        export default function() {};
+      `,
+      'b.ts': `
+        import './c';
+      `,
+      'c.ts': `
+        import './d';
+      `,
+      'd.ts': `
+        import './e';
+      `,
+      'e.ts': `
+        import './f';
+      `,
+      'f.ts': `
+        import './g';
+      `,
+      'g.ts': `
+        import './b';
+      `,
+    });
+    expect(context.modules).toHaveLength(8);
+    // we only have 1 split
+    expect(context.splitEntries.entries).toHaveLength(1);
+    // the split we have has 7 modules: a, b, c, d, e, f, g
+    expect(context.splitEntries.entries[0].modules).toHaveLength(7);
+  });
+});
```

Comment:
I would add a few cases with `require` and `export {foo} from "./foo"` just in case ;-)

Reply:
Those are already covered in the Import tests, but i'll add them for 'just in case' ;)

Reply:
✅ added extra test with different import / require types